### ISSUE TITLE
Regex.Escape and forced unicode flag

### ISSUE THREAD

**Issue by may-day** [fable-compiler/Fable#3325]
### Description

In the fable-library/RegExp.ts module the `unicode` flag is forced when creating a new regular expression.
But in the `escape` function the '-' char is [escaped](https://github.com/fable-compiler/Fable/blob/32592c46685a1f0489f902959eb8591b44551b7a/src/fable-library/RegExp.ts#L23).
This combination throws an:
```
Uncaught SyntaxError: Invalid regular expression: /\-/: Invalid escape
```
### Repro code
fable.io/repl is down, so i guess this should show
```fsharp
"-" |> Regex.Escape |> Regex
```
or in node:
```javascript
new RegExp("\\-", "u")
```

Without the unicode flag this works
### Expected and actual results

Expected: not throwing an error :)
Actual: App doesn't accept any more input/app hangs. Stepping through the code and manually testing in node yields the Error:
Uncaught SyntaxError: Invalid regular expression: /\-/: Invalid escape

### Related information

* Fable version: 4.0.0-theta-018
